### PR TITLE
[batch] remove endpoints for deleting and cancelling jobs

### PR DESCRIPTION
**Pull request by jigold**
It doesn't make sense to be able to delete or cancel an individual job since they must be part of a batch now. I also deleted `list_jobs` since a job must be a part of a batch. I left in `job.wait()` because I felt the tests in `test_dag` were important and shouldn't be deleted and the wait functionality is needed for there not to be race conditions.

### CODE REVIEW

File: batch/test/test_dag.py [hunk 1]
```diff
@@ -141,32 +111,15 @@ def test_cancel_left_after_tail(client):
...
-
 def test_one_of_two_parent_ids_cancelled(client):
     batch = client.create_batch()
     left = batch.create_job(
         'alpine:3.8',
         command=['/bin/sh', '-c', 'while true; do sleep 86000; done'])
     right = batch.create_job('alpine:3.8', command=['echo', 'right'])
     tail = batch.create_job('alpine:3.8', command=['echo', 'tail'], parent_ids=[left.id, right.id])
-    left.cancel()
+    right.wait()
```
Comment (akotlar) [inline batch/test/test_dag.py]: By not waiting on left do you introduce a race condition? Conversely, do you need to wait on right?

Comment (akotlar) [reply batch/test/test_dag.py]: Namely in other tests we’re waiting on left, right, head I consistently. We should either make them consistent I think, or explicitly indicate we’re testing for race conditions.

Comment (jigold) [reply batch/test/test_dag.py]: We don't want to wait on left here. It's going to sleep for a long time. The point of this test is to test what happens when left is cancelled but right is done already to make sure the child is also cancelled.

Comment (akotlar) [reply batch/test/test_dag.py]: this test no longer does that. You’re testing what happens when right is done, left is ? and the final parent (the batch) is cancelled.

File: batch/test/test_batch.py [hunk 2]
```diff
@@ -214,13 +183,9 @@ def test_batch(self):
         j2 = b.create_job('alpine', ['sleep', '1'])
         j3 = b.create_job('alpine', ['sleep', '30'])
 
-        # test list_jobs
-        jobs = self.batch.list_jobs()
-        self.assertTrue(
-            set([j.id for j in jobs]).issuperset([j1.id, j2.id, j3.id]))
-
+        j1.wait()
```
Comment (akotlar) [inline batch/test/test_batch.py]: Shouldn’t all calls to the batch raise if the batch is cancelled?

Comment (jigold) [reply batch/test/test_batch.py]: The batch is cancelled later. I don't see the issue.

File: batch/test/test_dag.py (hunk 1 above)
Comment (jigold) [reply batch/test/test_dag.py]: I'll just take out this test.

**Comment by jigold** [comment]
Closing in favor of future PR.

**Pull request closed without merging**